app/models/concerns/language_and_schedule_references.rb

```diff
@@ -5,6 +5,18 @@ module LanguageAndScheduleReferences
     has_many :language_skills, as: :languageable, dependent: :destroy
     accepts_nested_attributes_for :language_skills, allow_destroy: true
 
+    def native_language
+      return if language_skills.blank? || language_skills.native.blank?
+      language_skills.native.german.first || language_skills.native.first || LanguageSkill.new
+    end
+
+    def other_languages
+      return language_skills if language_skills.blank? || language_skills.native.blank?
+      language_skills.reject do |lang|
```

Comment:
Hmm, then you could return `LanguageSkill.new` instead of `nil` in `#native_language`. Or use ruby's safe operator.